benchmarks/solubility/plugin/solubility.cc

```diff
@@ -0,0 +1,409 @@
+/*
+  Copyright (C) 2014 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/melt.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <array>
+#include <utility>
+#include <limits>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * Volatiles material model.
+     * @ingroup MaterialModels
+     */
+    template <int dim>
+    class Volatiles : public MaterialModel::MeltInterface<dim>, public ::aspect::SimulatorAccess<dim>, public MaterialModel::MeltFractionModel<dim>
+    {
+      public:
+        /**
+         * Initialize the base model at the beginning of the run.
+         */
+        void initialize() override;
+
+        /**
+         * Update the base model at the beginning of each timestep.
+         */
+        void update() override;
+
+        /**
+         * Function to compute the material properties in @p out given the
+         * inputs in @p in.
+         */
+        void
+        evaluate (const typename Interface<dim>::MaterialModelInputs &in,
+                  typename Interface<dim>::MaterialModelOutputs &out) const override;
+
+        /**
+         * Method that indicates whether material is compressible. The model is compressible
+         * if and only if base model is compressible.
+         */
+        bool is_compressible () const override;
+
+        /**
+         * Compute the equilibrium melt fractions for the given input conditions.
+         * @p in and @p melt_fractions need to have the same size.
+         *
+         * @param in Object that contains the current conditions.
+         * @param melt_fractions Vector of doubles that is filled with the
+         * equilibrium melt fraction for each given input conditions.
+         */
+        virtual void melt_fractions (const MaterialModel::MaterialModelInputs<dim> &in,
```

Comment:
This I cannot change because it is a function that is implemented in the `MeltFractionModel` (one of the classes that this class is derived from) with this name. But I am changing the documentation.